scico/loss.py

```diff
@@ -140,7 +147,7 @@ def __call__(self, x: Union[JaxArray, BlockArray]) -> float:
         Args:
             x : Point at which to evaluate loss.
         """
-        return self.scale * self.functional(self.y - self.A(x))
+        return self.scale * (snp.abs(self.y - self.A(x)) ** 2).sum()
 
```

Comment:
Is this change desirable? Dropping the use if `self.functional` seems like it may have consequences for derived classes.

Reply:
I don't think it does. Does it?

I have this to keep it consistent with the weighted case where we can avoid computing the square root of the weights.

Reply:
Understood, and that's a worthwhile goal, but we should make sure there aren't any undesirable consequences before we make this change.

Reply:
I don't mind this change for the time being.

Longer term, consider: Why is `functional` a property at all if it is not used here? Broadly, I think this discussion is a symptom of the existence of `Loss`. One might hope to implement `__call__` at the level of `Loss` (and therefore using `self.functional`) to reduce repeated code. But we can't do that, because `Loss` is so general it doesn't know that it should be `A@x - y`.

Reply:
Agreed: if this change is made, we should consider removing the `functional` attribute. With respect to `Loss`, do you recall why it's so general? If there's good reason, perhaps we should have a specialization that really is `A@x - y`?

Reply:
I think the reason it is general is because of the Poisson. I think the reason it exists at all is that it used to not be a subclass of `Functional` and therefore having a base `Loss` class made sense.